st2actions/tests/unit/test_parallel_ssh.py

```diff
@@ -49,6 +51,8 @@ def test_connect_with_password(self):
             client._hosts_client[host].client.connect.assert_called_once_with(**expected_conn)
 
     @patch('paramiko.SSHClient', Mock)
+    @patch.object(ParamikoSSHClient, '_is_key_file_needs_passphrase',
```

Comment:
Needing to patch this "implementation detail" everywhere is kinda nasty and hard to maintain - is there no other way to achieve that?

Anyways, thanks for adding the test :+1: